### PR TITLE
utils: Decode bytecode into human-readable string

### PR DESCRIPTION
**Pull request by chfast**
Adds a procedure to convert a bytecode into a human-readable bytecode expression which is also compilable  back to bytecode with bytecode helpers.

TODO:
- [x] Add basic unit tests.

### CODE REVIEW

File: test/utils/bytecode.hpp [hunk 1]
```diff
@@ -163,4 +163,35 @@ inline bytecode sstore(bytecode index, bytecode value)
...
+        if (const auto name = names[opcode]; name)
+        {
+            s += std::string{" + OP_"} + name;
+
+            if (opcode >= OP_PUSH1 && opcode <= OP_PUSH32)
+            {
+                auto push_data = std::string{};
+                auto push_data_size = opcode - OP_PUSH1 + 1;
+                while (push_data_size-- && ++it != bytecode.end())
+                    push_data += to_hex({&*it, 1});
```
Comment (gumb0) [inline test/utils/bytecode.hpp]: Would be nice to have `to_hex` overload for a single byte parameter

Comment (gumb0) [reply test/utils/bytecode.hpp]: On the other hand, couldn't you do something like
```
auto push_data_start = ++it;
push_data_size = std::min(push_data_size, bytecode.end() - push_data_start);
push_data += to_hex({&*push_data_start, push_data_size});
it += push_data_size;
```
maybe too complicated...

File: test/unittests/bytecode_test.cpp [hunk 2]
```diff
@@ -48,3 +48,27 @@ TEST(bytecode, repeat)
...
+    const auto code1 = bytecode{} + OP_PUSH2 + "0000";
+    EXPECT_EQ(decode(code1, EVMC_FRONTIER), "bytecode{} + OP_PUSH2 + \"0000\"");
+
+    const auto code2 = bytecode{} + OP_PUSH2 + "00";
+    EXPECT_EQ(decode(code2, EVMC_FRONTIER), "bytecode{} + OP_PUSH2 + \"00\"");
+
+    const auto code3 = bytecode{} + OP_PUSH2;
+    EXPECT_EQ(decode(code3, EVMC_FRONTIER), "bytecode{} + OP_PUSH2");
+
+    const auto code4 = bytecode{} + OP_PUSH2 + "";
```
Comment (gumb0) [inline test/unittests/bytecode_test.cpp]: How is this different from `code3` ?

Comment (chfast) [reply test/unittests/bytecode_test.cpp]: Not really different. Just wanted to show the `""` will be dropped. In previous implementation `""` was added for every push with empty data.

File: test/utils/bytecode.hpp (hunk 1 above)
Comment (chfast) [reply test/utils/bytecode.hpp]: Great change actually. Allowed me to remove the additional iterator end check.

**Pull request merged**